### PR TITLE
Refactor ammo kits and crafting station

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

What it says on the tin. Most bugs and RNGs have been removed in the process.
Goodbye, sheets of -120 plasteel, you won't be missed.

Ammo kits now able to craft most magazine types.
Magazines spawn empty, with the exception of speed loaders.

Ammo kit initially get 15 "material points", that player could use to pick what they wanna craft.
MEC stat increases "material points", up to plus 15 at skill level 80.
Gunsmith perk gives additional 15.

Crafting station uses ammo kit system to craft ammunition.
Crafting station's components contribute to "material points", up to 10 extra points.
Crafting station's initial storage capacity halved, but benefits more from better components.
Also crafting station's animations isn't derpy anymore.

<details>
<summary>
"Price list"
</summary>

.20 ammo pile (10 ammo, 3 points)
.20 standard magazine (empty, 25 ammo, 5 points)
.20 extended magazine (empty, 35 ammo, 10 points)
.20 drum magazine (empty, 60 ammo, 15 points)
.20 ammo box (50 ammo, 15 points)

.25 ammo pile (10 ammo, 3 points)
.25 standard magazine (empty, 30 ammo, 5 points)
.25 ammo box (60 ammo, 15 points)

.30 ammo pile (10 ammo, 3 points)
.30 standard magazine (empty, 30 ammo, 5 points)
.30 ammo box (60 ammo, 15 points)

.35 ammo pile (15 ammo, 5 points)
.35 speedloader (6 ammo, 3 points)
.35 standard magazine (empty, 10 ammo, 5 points)
.35 extended magazine (empty, 16 ammo, 10 points)
.35 SMG magazine (empty, 35 ammo, 12 points)
.35 ammo box (70 ammo, 15 points)

.40 ammo pile (6 ammo, 3 points)
.40 speedloader (6 ammo, 5 points)
.40 standard magazine (empty, 10 ammo, 5 points)
.40 SMG magazine (empty, 25 ammo, 12 points)
.40 ammo box (30 ammo, 15 points)

.50 slug pile (5 ammo, 3 points)
.50 pellet pile (5 ammo, 3 points)
.50 beanbag pile (5 ammo, 3 points)
.50 slug box (30 ammo, 15 points)
.50 pellet box (30 ammo, 15 points)
.50 beanbag box (30 ammo, 15 points)

.60 ammo pile (5 ammo, 5 points)
.60 ammo box (30 ammo, 30 points)

</details>

<details>
<summary>
Screenshots
</summary>

image
image
image
image
image
image
image
image
image
image
image
image

</details>

<details>
<summary>
Videos
</summary>

https://user-images.githubusercontent.com/65828539/167271141-6820c4cd-2133-4f4f-bcc4-1f1853d2d111.mp4

https://user-images.githubusercontent.com/65828539/167271142-e871ac2b-7886-4b8a-afbc-8010c482d796.mp4

</details>

## Why It's Good For The Game

Old implementation have been horrible, both on the inside and on the outside.

## Changelog
:cl:
refactor: ammo kits and crafting station
/:cl:

### CODE REVIEW

**Comment by Humonitarian** [comment]
.70? You can print gyrojet bullets?

**Comment by SirRichardFrancis** [comment]
> .70? You can print gyrojet bullets?

Bruh. It meant to be antimaterial .60, let me correct that really quick.

**Comment by assortedbeads** [comment]
Looks solid, I dig not having to rng hope for a mag and instead getting loose ammo.

**Review by TheShown911 — APPROVED** [review]

File: code/game/machinery/crafting_station.dm [hunk 1]
```diff
@@ -67,560 +55,145 @@
...
 
-	var/man_rating = 0
-	for(var/obj/item/stock_parts/manipulator/M in component_parts)
-		man_rating += M.rating
 
-	var/las_rating = 0
-	for(var/obj/item/stock_parts/micro_laser/M in component_parts)
-		las_rating += M.rating
+/obj/machinery/craftingstation/RefreshParts()
+	productivity_bonus = initial(productivity_bonus)
```
Comment (SirRichardFrancis) [inline code/game/machinery/crafting_station.dm:179]:
```suggestion
	productivity_bonus = 0
```

Comment (SirRichardFrancis) [inline code/game/machinery/crafting_station.dm:179]:
```suggestion
	productivity_bonus = 0
```

**Pull request merged**